lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -249,6 +278,39 @@ bool ArrayAllocation::collectForwardableValues() {
 namespace {
 
 class ArrayElementPropagation : public SILFunctionTransform {
+
+  FuncDecl* lookupAppendFnDecl() {
```

Comment:
I think this belongs in `ASTContext.cpp` along with the rest of the stdlib lookup stuff.  We can also cache it there too to make repeated calls easier and knock out a parameter to the function that uses it.